### ISSUE TITLE
Sitemap requires date field

### ISSUE THREAD

**Issue by mankoff** [getpelican/pelican-plugins#229]
I have pages w/o `Date:` in the header. In this case the sitemap crashes now that I've upgrade to 3.4.0.

```
CRITICAL: expected string or buffer
Traceback (most recent call last):
  File "/Users/mankoff/Library/Enthought/Canopy_64bit/User/bin/pelican", line 9, in <module>
    load_entry_point('pelican==3.3', 'console_scripts', 'pelican')()
  File "/Users/mankoff/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pelican/__init__.py", line 419, in main
    pelican.run()
  File "/Users/mankoff/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pelican/__init__.py", line 173, in run
    p.generate_output(writer)
  File "/Users/mankoff/projects/www/kenmankoff.com/plugin/sitemap/sitemap.py", line 186, in generate_output
    self.set_url_wrappers_modification_date(self.context['categories'])
  File "/Users/mankoff/projects/www/kenmankoff.com/plugin/sitemap/sitemap.py", line 171, in set_url_wrappers_modification_date
    modified = self.get_date_modified(article, datetime.min);
  File "/Users/mankoff/projects/www/kenmankoff.com/plugin/sitemap/sitemap.py", line 161, in get_date_modified
    return get_date(getattr(page, 'modified'))
  File "/Users/mankoff/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pelican/utils.py", line 206, in get_date
    string = re.sub(' +', ' ', string)
  File "/Users/mankoff/local/Applications/Canopy.app/appdata/canopy-1.4.0.1938.macosx-x86_64/Canopy.app/Contents/lib/python2.7/re.py", line 151, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or buffer
```

Originally reported at: https://github.com/getpelican/pelican/issues/1389

**Comment by justinmayer** [comment]
Not sure if this is still a problem. If so, please feel free to comment here so the issue can be re-opened.

**Comment by jorgesumle** [comment]
@justimmayer, `<lastmod>` is optional according to sitemaps.org and, apparently, the sitemaps plugin is based on that specification as it links to it in the README.

![bildschirmfoto vom 2017-03-23 19-35-58](https://cloud.githubusercontent.com/assets/14212780/24264337/61b1ba34-1000-11e7-8eaa-6475d810013c.png)

**Comment by justinmayer** [comment]
@jorgesumle: So you're saying that the Sitemap plugin still needs work? If so, would you be willing to help address any shortcomings you might perceive?

**Comment by jorgesumle** [comment]
@justinmayer, I just tested it and it works perfectly with Pelican 3.7.1. But Pelican requires a 'date:' metadata, so the issue here was a configuration issue, I guess. If you don't want to specify the date metadata explicitly, you must set DEFAULT_DATE to 'fs', which relies on the file system.

> @jorgesumle: So you're saying that the Sitemap plugin still needs work?

Sorry, I was confused because I saw no reason to close the issue at first. It **should remain closed**.
